functions/apiProxy/participant_data_handler.py

```
@@ -19,6 +20,64 @@
+        
+    Returns:
+        Internal course ID (integer) or None if not found
+    """
+    # Try to parse as integer (internal ID)
+    try:
+        internal_id = int(course_id_param)
+        # Verify it exists in org_courses
+        for course in org_courses:
+            if course.get("_internalId") == internal_id or course.get("id") == internal_id:
+                return internal_id
+        return None
```

Ruff (0.14.7)

Consider moving this statement to an `else` block

(TRY300)